reposerver/src/test/scala/com/advancedtelematic/tuf/reposerver/http/RepoResourceDelegationsSpec.scala

```diff
@@ -167,4 +167,72 @@ class RepoResourceDelegationsSpec extends TufReposerverSpec
       responseAs[ErrorRepresentation].code shouldBe ErrorCodes.PayloadSignatureInvalid
     }
   }
+
+  test("re-generates snapshot role after storing delegations") {
+    implicit val repoId = addTargetToRepo()
+
+    addDelegationToRepo()
+
+    Get(apiUri(s"repo/${repoId.show}/snapshot.json")) ~> routes ~> check {
+      status shouldBe StatusCodes.OK
+      responseAs[SignedPayload[SnapshotRole]].signed.version shouldBe 2
+    }
+  }
+
+  test("SnapshotRole includes signed delegation length") {
+    implicit val repoId = addTargetToRepo()
+
+    addDelegationToRepo()
+
+    val signedDelegationRole = buildSignedDelegatedTargets()
+
+    pushSignedDelegatedMetadata(signedDelegationRole) ~> check {
+      status shouldBe StatusCodes.NoContent
+    }
+
+    val delegationRole =
+      Get(apiUri(s"repo/${repoId.show}/delegations/${delegatedRoleName.value}.json")) ~> routes ~> check {
+        status shouldBe StatusCodes.OK
+        responseAs[SignedPayload[TargetsRole]]
+      }
+
+    val delegationLength = delegationRole.asJson.canonical.length
+
+    Get(apiUri(s"repo/${repoId.show}/snapshot.json")) ~> routes ~> check {
+      status shouldBe StatusCodes.OK
+      val signed = responseAs[SignedPayload[SnapshotRole]].signed
+
+      signed.meta(Refined.unsafeApply(s"${delegatedRoleName.value}.json")).length shouldBe delegationLength
+    }
+  }
+
+  test("automatically renewed snapshot still contains delegation") {
```

Comment:
I think this can maybe be simplified by just comparing the renewed snapshot is the same except the version/expire date, we might cover that with other tests already

Reply:
I don't think so, I think this tests a different path of SignedRoleGeneration.findFreshRole(), one path the refreshing and the other the first creation of a role (what the underlying ticket is about). I think there is some refactoring to do in general, regenerateAllSignedRoles is similar to RepoRoleRefresh.refreshTargets, e.g.